service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendAcknowledgementExecutor.java

```diff
@@ -0,0 +1,64 @@
+package uk.nhs.adaptors.gp2gp.ehr;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.mustachejava.Mustache;
+
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+import uk.nhs.adaptors.gp2gp.common.service.TimestampService;
+import uk.nhs.adaptors.gp2gp.common.task.TaskExecutor;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+import uk.nhs.adaptors.gp2gp.mhs.MhsClient;
+import uk.nhs.adaptors.gp2gp.mhs.MhsRequestBuilder;
+import uk.nhs.adaptors.gp2gp.mhs.model.OutboundMessageWithPayload;
+
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Service
+public class SendAcknowledgementExecutor implements TaskExecutor<SendAcknowledgementTaskDefinition> {
+    private static final Mustache ACKNOWLEDGEMENT_TEMPLATE =
+        TemplateUtils.loadTemplate("outbound_message_positive_acknowledgement.mustache");
+    private final MhsClient mhsClient;
+    private final MhsRequestBuilder mhsRequestBuilder;
+    private final EhrExtractStatusService ehrExtractStatusService;
+    private final ObjectMapper objectMapper;
+    private final TimestampService timestampService;
+    private final RandomIdGeneratorService randomIdGeneratorService;
+
+    @Override
+    public Class<SendAcknowledgementTaskDefinition> getTaskType() {
+        return SendAcknowledgementTaskDefinition.class;
+    }
+
+    @Override
+    @SneakyThrows
+    public void execute(SendAcknowledgementTaskDefinition sendAcknowledgementTaskDefinition) {
+        LOGGER.info("SendAcknowledgement task was created, Sending Acknowledgement to Spine");
+        var messageId = randomIdGeneratorService.createNewId();
```

Comment:
This is not a random identifier. AC3 says it must be the message id of the original EhrRequest message.

Reply:
I'd also considering renaming any references to this particular messageId ehrRequestMessageId to avoid confusion with the id of the message we're about to generate